eng/common/scripts/job-matrix/README.md

```diff
@@ -56,6 +56,7 @@ jobs:
         Cloud: Public
       MatrixFilters: []
       MatrixReplace: []
+      PreGenerationSteps: []
```

Comment:
I can put a simple

> Write-Host gci $(ClonedRepo) 

Or something if we want to have an example of this in the methodolgoy. We don't have that for the "further" features like MatrixFilters or MatrixReplace though, so I feel like we're probably fine.

Reply:
Or a section in the docs regarding dynamic matrix generation in a pipeline: https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/README.md#usage-in-a-pipeline

Reply:
My paranoia panned out. This doesn't work.

Reply:
@benbp added a readme section as requested.